packages/paste-core/components/side-panel/src/SidePanel.tsx

```diff
@@ -31,87 +61,145 @@ const config = {
   friction: 20,
 };
 
-const transitionStyles = {
-  from: { opacity: 0, transform: "translateX(100%)" },
-  enter: { opacity: 1, transform: "translateX(0%)" },
-  leave: { opacity: 0, transform: "translateX(100%)" },
-  config,
-};
-
-const mobileTransitionStyles = {
-  from: { opacity: 0, transform: "translateY(100%)" },
-  enter: { opacity: 1, transform: "translateY(0%)" },
-  leave: { opacity: 0, transform: "translateY(100%)" },
-  config,
-};
-
-const SidePanel = React.forwardRef<HTMLDivElement, SidePanelProps>(
-  ({ element = "SIDE_PANEL", label, children, ...props }, ref) => {
-    const { sidePanelId, isOpen } = React.useContext(SidePanelContext);
-
-    const { breakpointIndex } = useWindowSize();
-
-    const transitions =
-      breakpointIndex === 0 ? useTransition(isOpen, mobileTransitionStyles) : useTransition(isOpen, transitionStyles);
-
-    const screenSize = window.innerHeight;
+interface SidePanelContentsProps extends SidePanelProps {
+  sidePanelId: string;
+  styles: any;
+}
 
+const SidePanelContents = React.forwardRef<HTMLDivElement, SidePanelContentsProps>(
+  ({ label, element, sidePanelId, styles, children, ...props }, ref) => {
+    // Get the offset of the side panel from the top of the viewport
     const sidePanelRef = React.useRef<HTMLDivElement>(null);
     const mergedSidePanelRef = useMergeRefs(sidePanelRef, ref) as React.RefObject<HTMLDivElement>;
-
+    const screenSize = window.innerHeight;
```

Comment:
wouldn't the `window` object be undefined outside useEffect?

Reply:
this isn't new logic it just moved down a bit - should be working